nilearn/_utils/niimg_conversions.py

```diff
@@ -271,6 +272,34 @@ def check_niimg(niimg, ensure_ndim=None, atleast_4d=False, dtype=None,
                 raise ValueError("File not found: '%s'" % niimg)
         elif not os.path.exists(niimg):
             raise ValueError("File not found: '%s'" % niimg)
+    
+    if isinstance(niimg, pathlib.Path):
+        # get the string from pathlib object
+        # and then process like a normal string
+        niimg = niimg.absolute().as_posix()
```

Comment:
Unless I missed something, the code inside this if statement is exactly the same as the code just above. 
One much simpler way to rewrite this would be to place this if at the beginning of the function (which will cast niimg to a string if it is a Path), and let the previous block of code handle the rest.
Does this make sense?

Reply:
Yes, will update.

Reply:
why take the absolute path and why `as_posix` ?